rules_bazel/java/java_integration_test.bzl

```diff
@@ -0,0 +1,82 @@
+def _compare_with_goldens_test_impl(ctx):
+    diff_output = ctx.outputs.diff_output
+    check_diff_script = ctx.outputs.check_diff_script
+    input = ctx.attr.api_target
+    srcs = ctx.files.srcs
+    api_name = ctx.attr.name
+    
+    script = """
+    mkdir codegen_tmp
+    unzip -j {input} -d codegen_tmp
+    unzip -j {input_resource_name} -d  codegen_tmp
+    unzip -j {input_test} -d codegen_tmp
+    cd codegen_tmp
+    # Remove unneeded non-Java files, like MANIFEST
+    rm -rf $(find . -type f ! -name "*.java")
+    cd ..
+    diff codegen_tmp test/integration/goldens/{api_name}/ > {diff_output}
+    """.format(
+        diff_output = diff_output.path,
+        input = input[0][JavaInfo].source_jars[0].path,
```

Comment:
If you prefer accessing each individual source jar by its index here, then pelase consider just splitting  this one api_target list argument into 3 different arguments, each of which with a descriptive name, something like:
`gapic_library`
`resource_name_library`
`test_library`

Reply:
Sure, I think that would make the input more descriptive. Renamed them, thank you!

Reply:
@miraleung previously we have only 1 arg (target) which we can remove the wrapper function of `integration_test`, while adding a wrapper to call `compare_with_goldens_test` can help us split the list arguments and give a descriptive name. WDYT?